deeppavlov/core/trainers/fit_trainer.py

```diff
@@ -113,32 +133,37 @@ def fit_chainer(self, iterator: Union[DataFittingIterator, DataLearningIterator]
                     writer = None
 
                     for i, (x, y) in enumerate(iterator.gen_batches(self.batch_size, shuffle=False)):
-                        preprocessed = self._chainer.compute(x, y, targets=targets)
+                        preprocessed = self._chainer.compute(
+                            x, y, targets=targets)
                         # noinspection PyUnresolvedReferences
                         result = component.partial_fit(*preprocessed)
 
-                        if result is not None and self.tensorboard_log_dir is not None:
+                        if result is not None and self.tensorboard_idx is not None:
                             if writer is None:
-                                writer = self._tf.summary.FileWriter(str(self.tensorboard_log_dir /
+                                writer = self._tf.summary.FileWriter(str(self.logger[self.tensorboard_idx]["log_dir"] /
                                                                          f'partial_fit_{component_index}_log'))
                             for name, score in result.items():
                                 summary = self._tf.Summary()
-                                summary.value.add(tag='partial_fit/' + name, simple_value=score)
+                                summary.value.add(
+                                    tag='partial_fit/' + name, simple_value=score)
                                 writer.add_summary(summary, i)
                             writer.flush()
                 else:
-                    preprocessed = self._chainer.compute(*iterator.get_instances(), targets=targets)
+                    preprocessed = self._chainer.compute(
+                        *iterator.get_instances(), targets=targets)
                     if len(targets) == 1:
                         preprocessed = [preprocessed]
-                    result: Optional[Dict[str, Iterable[float]]] = component.fit(*preprocessed)
+                    result: Optional[Dict[str, Iterable[float]]
+                                     ] = component.fit(*preprocessed)
 
-                    if result is not None and self.tensorboard_log_dir is not None:
```

Comment:
```
if result is not None:
    for lgr in self.logger:
        lgr.log_fit(result)
```
And add log_fit method to base logger class. TensorboardLogger realization of this method should contain following code with writer and summary